dask/dataframe/io/tests/test_io.py

```diff
@@ -496,15 +496,29 @@ def test_to_bag():
     assert ddf.x.to_bag().compute() == list(a.x)
 
 
-def test_to_records():
+@pytest.mark.parametrize('lengths', [
+    None,
+    [2, 3],
+    True,
+])
+def test_to_records(lengths):
     pytest.importorskip('dask.array')
     from dask.array.utils import assert_eq
     df = pd.DataFrame({'x': ['a', 'b', 'c', 'd'],
                        'y': [2, 3, 4, 5]},
                       index=pd.Index([1., 2., 3., 4.], name='ind'))
     ddf = dd.from_pandas(df, 2)
 
-    assert_eq(df.to_records(), ddf.to_records())
+    assert_eq(df.to_records(), ddf.to_records(lengths=lengths))
+
+    # assert isinstance(result, da.Array)
+    #
+    # expected_chunks = ((2, 3),)
+    #
+    # if as_frame:
+    #     expected_chunks = expected_chunks + ((1,),)
+    #
+    # assert result.chunks == expected_chunks
```

Comment:
Add some tests for the cases that raise as well

1. len(lenghts) != npartitions
2. an invalid value for lengths.

Reply:
In progress (hopefully this week, possibly next).